testDataSetup/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,182 @@
+package elasticsearch
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/rchttp"
+)
+
+type Index struct {
+	InstanceID   string
+	Dimension    string
+	TestDataFile string
+	URL          string
+	MappingsFile string
+}
+
+var client = rchttp.DefaultClient
+
+// ErrorUnexpectedStatusCode represents the error message to be returned when
+// the status received from elastic is not as expected
+var ErrorUnexpectedStatusCode = errors.New("unexpected status code from elasticsearch api")
+
+// DeleteIndex removes a specified index from elasticsearch
+func DeleteIndex(path string) (int, error) {
+	logData := log.Data{"url": path}
+
+	URL, err := url.Parse(path)
+	if err != nil {
+		log.ErrorC("failed to create url for elasticsearch call", err, logData)
+		return 0, err
+	}
+	path = URL.String()
+
+	req, err := http.NewRequest("DELETE", path, nil)
+	if err != nil {
+		log.ErrorC("failed to create request for call to elasticsearch", err, logData)
+		return 0, err
+	}
+
+	resp, err := client.Do(context.Background(), req)
+	if err != nil {
+		log.ErrorC("failed to call elasticsearch", err, logData)
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	logData["http_code"] = resp.StatusCode
+	if resp.StatusCode != http.StatusOK {
+		log.Error(ErrorUnexpectedStatusCode, logData)
+		return resp.StatusCode, ErrorUnexpectedStatusCode
+	}
+
+	return http.StatusOK, nil
+}
+
+// Dimension represents the data stored in an elasticsearch document
+type Dimension struct {
+	Code             string `json:"code"`
+	HasData          bool   `json:"has_data"`
+	Label            string `json:"label"`
+	NumberOfChildren int    `json:"number_of_children"`
+	URL              string `json:"url"`
+}
+
+// CreateSearchIndex represents the creation and loading of test data into an index for testing
+func (i *Index) CreateSearchIndex() error {
+	index := i.URL + "/" + i.InstanceID + "_" + i.Dimension
+	// Remove index
+	statusCode, err := DeleteIndex(index)
+	if err != nil {
+		if statusCode != http.StatusNotFound {
+			log.ErrorC("failed to delete index", err, log.Data{"path": index})
+			return err
```

Comment:
Should this trigger `os.Exit(1)`? So its like the mongodb setup

Reply:
No the thing calling this method would need to handle the error